File: Functional/Functional/Functional/FunctionalViewModel.cs
using Functional.Base;
using Functional.Extensions;
using Xamarin.Forms;

namespace Functional
{
	public class FunctionalViewModel : DefineViewModel<ViewState, Controller> { }

	public class ViewState : BaseViewState
	{
		public string LabelValue { get; set; }
	}

	public class Controller : BaseController
	{
		public Command<Data[]> UpdateCommand => new Command<Data[]>((data) =>
		{
			var request = new Request()
			{
				{ nameof(ViewState.LabelValue), PureFunction(data[0].Value, 
															 data[1].Value) }
			};

			Push(request);
		});
		
		string PureFunction(string firstString, string lastString)
		{
			return $"{firstString} {lastString}";
		}
	}

}

File: FunctionalFSharp/FunctionalViewModel.fs
// This is Revision #2

namespace FunctionalFSharp

open Functional.Base
open Xamarin.Forms
open Functional.Extensions
open System
open System.Collections.Generic

type View() =    
    inherit BaseViewState()
    
    member val LabelValue = "" with get, set   

type Controller() as this =
    inherit BaseController()

    let combine first last = first + " " + last
    
    let f (e: Data[]) = 
        let request = new Request()
        let first = e.[0].Value
        let second = e.[1].Value
        
        request.Add("LabelValue", combine first second)
        this.Push(request) |> ignore
    
    let command = new Command<_>(System.Action<_>(f))

    member val UpdateCommand = command

type FunctionalFSharpViewModel() =
    inherit DefineViewModel<View, Controller>()

